tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/event/RabbitMQAndRedisEventBusModule.java

```diff
@@ -81,10 +83,15 @@ sender, receiverProvider, eventSerializer, retryBackoffConfiguration, new Routin
     }
 
     @ProvidesIntoSet
-    InitializationOperation workQueue(RabbitMQAndRedisEventBus instance) {
+    InitializationOperation workQueue(RabbitMQAndRedisEventBus instance, CleanRedisEventBusService cleanRedisEventBusService) {
         return InitilizationOperationBuilder
             .forClass(RabbitMQAndRedisEventBus.class)
-            .init(instance::start);
+            .init(() -> {
+                instance.start();
+                cleanRedisEventBusService.cleanUp()
+                    .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
+                    .subscribe();
```

Comment:
Why?

 -> Is it ok to start the redis event bus while the previous one isn't well clean up?
 
 -> How long does this take? Is it necessary to do it asynchronously?
 
 Doing this with a `.block` in the init sequence looks safer and simpler to me.
 
 Thoughts?

Reply:
> Is it ok to start the redis event bus while the previous one isn't well clean up?

Yes, it is. Publish messages to the dangling bindings would not fail. I tested in the previous commits.

> How long does this take?

I can not say how long yet. Maybe I can measure it on staging tomorrow. My guess: a few seconds if the number of bindings is big.

> Is it necessary to do it asynchronously?

My idea when I put it asynchronous is to avoid causing more startup time for TMail, potentially reducing the downtime too. Given that JVM startup time is long already and especially TMail/James is fatter over time.

I am not sure how long the cleanup takes, especially when the data scales like millions of mailboxes, so I thought it would be safer (for startup time) to do it async.

> Doing this with a .block in the init sequence looks safer and simpler to me.

ATM I do log the result of the cleanup, with the number of bindings cleaned... So I think it is "safe" enough to know how the cleanup ran.